pkg/network/vmicontroller/status.go

```diff
@@ -51,6 +51,16 @@ func UpdateStatus(clusterConfig clusterConfigChecker, vmi *v1.VirtualMachineInst
 
 	interfaceStatuses = append(interfaceStatuses, secondaryIfaceStatuses...)
 
+	// virt-handler adds status for interfaces only existing inside the guest that were discovered by QEMU GuestAgent
+	guestInternalIfaceStatuses := vmispec.FilterInterfaceStatuses(
+		vmi.Status.Interfaces,
+		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
+			return ifaceStatus.Name == "" && vmispec.ContainsInfoSource(ifaceStatus.InfoSource, vmispec.InfoSourceGuestAgent)
```

Comment:
Is the `ifaceStatus.Name` empty when the `InfoSource` is `InfoSourceDomainAndGA`? I'm pretty sure we hit this case when the domain status iface name is populated.

Reply:
Also perhaps we should consider adding a test for this case specifically.

Reply:
https://github.com/kubevirt/kubevirt/blob/4e49259b9036478b2232c5ad3bab34a6103dc165/staging/src/kubevirt.io/api/core/v1/types.go#L695-L696

Since the interface is internal to the guest (does not exist in the VMI spec), per my understanding its ifaceStatus.Name should be empty.

Reply:
Fair enough.